### PR TITLE
Connect TLVM to in memory database

### PR DESCRIPTION
**Pull request by Ura3535**
Solves #39 and #41
- Add `AppServiceProvider` class for providing singleton dependencys.
- Create `repository` (as singleton dependency) in `TaskListViewModel`.
- All methods in `TaskListViewModel` have been changed to work with `repository`.
- Add `UpdateTaskStatus` method in `IRepository` (and add Unit test).
- Create `InitializeInMemoryDataBase` and move `GenerateRandomTask` to the Temporary folder.

### CODE REVIEW

File: App/Models/Task.cs [hunk 1]
```diff
@@ -2,16 +2,20 @@ namespace App.Models
 {
     public class Task
     {
-        public static readonly long DEFAULT_ID = -1;
+        public const long DEFAULT_ID = -1;
+        public const bool DONE = true;
```
Comment (WizardPlatypus) [inline App/Models/Task.cs:6]: I am not sure the `DONE` and `NOT_DONE` constants are at all necessary. I understand that they were added to get rid of "magic numbers", similarly to how `DEFAULT_ID` was introduced. But interacting with a field like `IsDone` using boolean values is very intuitive, and aliasing `true` and `false` to `DONE` and `NOT_DONE` doesn't provide any additional clarity.


I think these constants can be safely removed, as it's unlikely they are gonna be used very often.

File: App/Repositories/IRepository.cs [hunk 2]
```diff
@@ -2,14 +2,15 @@
...
+    public interface IRepository
     {
         long AddTask(Task newTask);
-        void DeleteTask(long task_id);
-        void UpdateTaskTitle(long task_id, string newTitle);
-        void UpdateTaskDescription(long task_id, string newDescription);
-        void UpdateTaskDueTime(long task_id, DateTimeOffset newDue);
-        Task GetTask(long task_id);
+        void DeleteTask(long taskId);
+        void CompleteTask(long taskId);
```
Comment (WizardPlatypus) [inline App/Repositories/IRepository.cs]: I woner if `CompleteTask` should be changed to `void UpdateTaskStatus(long task_id, bool isDone)`, to be consistent with other names of the kind, and to allow "un-doing" a task.

**Review by WizardPlatypus — CHANGES_REQUESTED** [review]
Good job!

I'm glad you were able to figure out a working dependency injection strategy, and I thank you for your other small refactorings.

I left 2 comments requesting changes. Once we resolve that, this is ready for merge!

File: App/Models/Task.cs (hunk 1 above)
Comment (Ura3535) [reply App/Models/Task.cs:6]: You're thinking exactly right, I added them to get rid of the magic numbers.
But I still think that the `IsDone` field should be changed to `Status` and stored as an enum

File: App/Repositories/IRepository.cs (hunk 2 above)
Comment (Ura3535) [reply App/Repositories/IRepository.cs]: As I said in issue #41, at this point we needed to add a method that changes `IsDone` to `true` (and later to `false`). I think I'll do that now.
By the way, should we throw an exception if we change the `IsDone` value back to what it was (we don't expect the completed task to be completed again)?

File: App/Models/Task.cs (hunk 1 above)
Comment (Ura3535) [reply App/Models/Task.cs:6]: And the call to `repository.UpdateTaskStatus(taskId, Task.DONE);`
is more understandable than `repository.UpdateTaskStatus(taskId, true);`

Comment (WizardPlatypus) [reply App/Models/Task.cs:6]: Agreed

File: App/Repositories/IRepository.cs (hunk 2 above)
Comment (WizardPlatypus) [reply App/Repositories/IRepository.cs]: I think an exception would be the wrong kind of response. We could instead make `CompleteTask`, or `UpdateTaskStatus`, to return the previous value of the field.

**Review by WizardPlatypus — APPROVED** [review]
If you wish to implement the `Status` enum or change `UpdateTaskStatus` to return the previous state of the field, you are welcome to do so. I will make sure to re-approve this pull request in a few hours if you do.

Otherwise, you can merge these changes.

File: App/Repositories/IRepository.cs (hunk 2 above)
Comment (Ura3535) [reply App/Repositories/IRepository.cs]: Ok, then I'll remove the exception throwing for now, and then we'll change it as needed

**Comment by Ura3535** [comment]
> If you wish to implement the `Status` enum or change `UpdateTaskStatus` to return the previous state of the field, you are welcome to do so. I will make sure to re-approve this pull request in a few hours if you do.
> 
> Otherwise, you can merge these changes.

Okay, I think we'll talk about Status enum today, as well as possible issues (such as completing the task that has been completed) and the best solution to fix them.
I will change this in another branch, and this one can be merged.

**Review by WizardPlatypus — APPROVED** [review]

**Pull request merged**